index.js

```diff
@@ -29,9 +30,45 @@ function sendBootStatus( status ) {
 	process.send( { boot: status } );
 }
 
-console.log( chalk.yellow( '%s booted in %dms - http://%s:%s' ), pkg.name, ( Date.now() ) - start, host, port );
+console.log(
+	chalk.yellow( '%s booted in %dms - %s://%s:%s' ),
+	pkg.name,
+	( Date.now() ) - start,
+	protocol,
+	host,
+	port
+);
 
-server = http.createServer( app );
+// Start a development HTTPS server.
+if ( protocol === 'https' ) {
+	const { execSync } = require( 'child_process' );
+	const execOptions = { encoding: 'utf-8', windowsHide: true };
+	let key;
+	let certificate;
+
+	try {
+		execSync( 'openssl version', execOptions );
+		execSync( 'openssl req -x509 -newkey rsa:2048 -keyout ./config/server/key.tmp.pem -out ./config/server/certificate.pem -days 365 -nodes -subj "/C=US/ST=Foo/L=Bar/O=Baz/CN=calypso.localhost"', execOptions );
+		execSync( 'openssl rsa -in ./config/server/key.tmp.pem -out ./config/server/key.pem', execOptions );
+		execSync( 'rm ./config/server/key.tmp.pem', execOptions );
+
+		key = './config/server/key.pem';
+		certificate = './config/server/certificate.pem';
```

Comment:
Can we check if the local `key.pem` and `certificate.pem` already exist before generating them? Currently, a new certificate is generated on every `npm start` and I need to confirm a security exception in the browser every time.

Other than that, the HTTPS server works 👌 Thanks for making it happen!

Reply:
Sure. Will do that on Monday :-).

Reply:
Done!